src/main/java/com/atlassian/jira/cloud/jenkins/buildinfo/service/FreestyleJiraBuildInfoSenderImpl.java

```diff
@@ -38,7 +39,8 @@ public FreestyleJiraBuildInfoSenderImpl(
     }
 
     @Override
-    protected Set<String> getIssueKeys(final JiraBuildInfoRequest request) {
+    protected Set<String> getIssueKeys(
+            final JiraBuildInfoRequest request, final PipelineLogger pipelineLogger) {
```

Comment:
Many changes in this PR are about passing a `PipelineLogger` into a method and using it to log additional log events.